runtime/jvmti/jvmtiHook.c

```diff
@@ -3041,16 +3051,15 @@ jvmtiHookCompilingEnd(J9HookInterface** hook, UDATA eventNum, void* eventData, v
 static void 
 jvmtiHookSampledObjectAlloc(J9HookInterface** hook, UDATA eventNum, void* eventData, void* userData)
 {
-	J9JVMTIEnv * j9env = userData;
+	J9JVMTIEnv *j9env = userData;
 	jvmtiEventSampledObjectAlloc callback = j9env->callbacks.SampledObjectAlloc;
+	J9VMThread *currentThread = ((MM_ObjectAllocationSamplingEvent *)eventData)->currentThread;
```

Comment:
Earlier, I had missed that `data` was also being used on line 3072. We will need a local variable. @thallium Your initial impl was correct.
```suggestion
	MM_ObjectAllocationSamplingEvent *data = eventData;
	J9VMThread *currentThread = data->currentThread;
```